cmake/external/dml.cmake

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+if (NOT onnxruntime_USE_CUSTOM_DIRECTML)
+  if (NOT(MSVC) OR NOT(WIN32))
+    message(FATAL_ERROR "NuGet packages are only supported for MSVC on Windows.")
+  endif()
+
+  # Retrieve the latest version of nuget
+  include(ExternalProject)
+  ExternalProject_Add(nuget
+    PREFIX nuget
```

Comment:
We can put the nuget part into CI build pipeline YAML files. 
If you prefer to leave them here, then It's better to set a fixed version here, so that the build is more reproducible .